src/Dashboard/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import books from './books';
+import './style.css';
+import logo from '../assets/search.svg';
+
+class BooksList extends Component {
+  render() {
+    const books = this.props.books;
+    const listItems = books.map((books) =>
+      <div key={books.id}>
+        <img alt="" className="book-logo" src={books.image_url} />
+        <h2 className="book-title">{books.title}</h2>
+        <p className="book-author">{books.author}</p>
+      </div>
+    );
+
+    return (
+      <div className="grid">{listItems}</div>
+    );
+  }
+}
+
+class Dashboard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      filter: ''
+    };
+
+    this.handleSelectChange = this.handleSelectChange.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleSelectChange(event) {
+    this.setState({ filter: event.target.value});
+  }
+
+  handleInputChange(event) {
+    this.setState({ value: event.target.value});
+  }
+
+  handleClick(event) {
+    if (event.key === 'Enter' | event.type === "click") {
+      var updatedList = books;
+      
+      if (this.state.filter !== '') {
+        updatedList = updatedList.filter (book => 
+            book[this.state.filter].toLowerCase().search(this.state.value.toLowerCase()) !== -1);
+      }
+
+      this.setState({items: updatedList});
```

Comment:
I think there is a better approach to this component.
In general you want to identify the minimal representation of state, in this view the things that change are:

1. The filter the user is typing.
2. The filter that was submitted.
3. The filtered books.

The thing is that if you have 1 and 2 in state you can calculate the filtered books with the books list (now it is a mock but later it will probably come from props) and the submitted filter state.

So you'll need 4 strings for the state (maybe call them filter, value, submittedFilter, submittedValue)
And then you can filter the books in render:
```
filteredBooks() {
  // uses this.state.submittedFilter, this.state.submittedValue and books to return the filtered list
}

render() {
  ...
  this.filteredBooks()
  ...
}
```

So it's not necessary to set the items in state.